lib/vtls/mbedtls.c

```diff
@@ -574,25 +574,25 @@ mbed_connect_step2(struct connectdata *conn,
 
   ret = mbedtls_ssl_get_verify_result(&BACKEND->ssl);
 
+  if(!SSL_CONN_CONFIG(verifyhost))
+    /* Ignore hostname errors if verifyhost is disabled */
+    ret &= ~MBEDTLS_X509_BADCERT_CN_MISMATCH;
+
   if(ret && SSL_CONN_CONFIG(verifypeer)) {
-    if(ret & MBEDTLS_X509_BADCERT_EXPIRED)
+    if(MBEDTLS_X509_BADCERT_EXPIRED)
```

Comment:
I'm pretty sure that this and the following 3 `if` expressions need to be `(ret & MBEDTLS_X509_BADCERT_xxxx)` since they're testing specific bits in `ret` to see why validation failed.

We know that it failed because at least one bit is set, these are just figuring out why for the sake of the `failf` call.

Reply:
Thanks, it was a silly move of me...